protected/humhub/modules/space/modules/manage/jobs/ChangeContentVisibilityJob.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) 2023 HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\space\modules\manage\jobs;
+
+use humhub\modules\content\models\Content;
+use humhub\modules\queue\ActiveJob;
+
+class ChangeContentVisibilityJob extends ActiveJob
+{
+    public int $contentContainerId;
+
+    public int $visibility;
+
+    public function run()
+    {
+        Content::updateAll(
```

Comment:
We need to loop through all the contents (With Public Visibility) here and run `$content->save()`. This is necessary so that e.g. search indexes are also updated.